### ISSUE TITLE
Add Block: Copyright

### ISSUE THREAD

**Issue by melchoyce** [WordPress/gutenberg#18895]
This block would be useful for themes to use once we have Full-site editing, and would automatically output the site title, ©, and the year. Additionally, it would be good if themes could somehow hook into it to add an author credit. Like:

> ©2019 [My Example Site]. Theme by [Theme Author].

**Comment by BinaryMoon** [comment]
I like to add an automatic link to the privacy policy in my theme footers too.

**Comment by richtabor** [comment]
> I like to add an automatic link to the privacy policy in my theme footers too.

100% this, based on the privacy policy WP page.

**Comment by mapk** [comment]
Initially, I thought this could be based largely on the Paragraph block. The block settings would include an option for the theme's author as a toggle which appends that text to whatever text the user wrote in the block.

**Block**

![Copyright](https://user-images.githubusercontent.com/617986/71494697-efb44780-27fd-11ea-86a7-0c0ac7e9942f.png)

**Block Settings**

![Block Settings](https://user-images.githubusercontent.com/617986/71494700-f3e06500-27fd-11ea-89bb-7ee4cf6265c4.png)

----

But after some thinking about this, what if made this a "License block"? And people can change between "Copyright" or "Creative Commons" or "GPL", whatever license we wanted to include. Do you think this makes sense?

Basically, when switching to a new license, the license icon swaps out to show one of the other lisence icons like `CC` for Creative Commons.

**Block**

![License-block](https://user-images.githubusercontent.com/617986/71495173-5850f380-2801-11ea-8d1e-c12096ffb55f.png)

**Block Settings**

![License-block-settings](https://user-images.githubusercontent.com/617986/71495180-5e46d480-2801-11ea-9ac5-b3f38bfc1da3.png)

**Comment by ZebulanStanphill** [comment]
@mapk In that second mockup, I think "License type" should not be in the inspector. I consider it to be a primary option of the block, and therefore it should be either in the toolbar or an inline control shown within the block's edit canvas.

It should also be noted that you would probably use the standard copyright symbol regardless of what license you are using.

**Comment by karmatosed** [comment]
I like the idea about the licence, I wonder though if it should be an option at the point of placeholder or something to switch. I can see a step for example of 'choose license' and it would create. I kind of agree it doesn't fit as well in the sidebar, but I more think it's a first step over having in toolbar. I do love a block that doesn't need sidebar options, they are simple and easier for people.

**Comment by mapk** [comment]
Really great feedback! Thank you. I wonder if we should start with a set of licenses we'd like to expose in this block?

- Copyright
- Creative Commons
-- Attribution
-- Attribution Share Alike
-- Attribution No Derivatives
-- Attribution Non-Commercial
-- Attribution Non-Commercial Share Alike
-- Attribution Non-Commercial No Derivatives 
- Public Domain
- [GNU Free Documentation License (GFDL)](http://www.gnu.org/licenses/fdl.html)

I'm not sure how well GPL aligns with just web content. Same question for other licenses like MIT and Apache. Should we include those options?

Are there others I haven't included?

**Comment by enriquesanchez** [comment]
In addition to the ability to add a license, it'll be super useful if the block auto updated the year as time passes.

**Comment by peroyomas** [comment]
Not sure if a new block is needed for this, but something that can help a lot is a **core shortcode that displays the current year**. Many themes has a similar solution to keep the copyright updated, so it may be good to feature it in core, too.

**Comment by pagelab** [comment]
@peroyomas There's[ already a proposal](https://github.com/WordPress/gutenberg/discussions/39831) for creating a sort of _shortcodes v2_ – or dynamic tokens – for use cases like this one.

**Comment by paaljoachim** [comment]
Hey

I happen to create a duplicate: https://github.com/WordPress/gutenberg/issues/47522
It shows some newer examples from current plugins.

**Comment by paaljoachim** [comment]
I will bring over a few comments to this issue from the newer issue that I made. https://github.com/WordPress/gutenberg/issues/47522
I closed it as a duplicate.

Michelle has made a Copyright year block located here: https://github.com/marktimemedia/copyright-year-block
I believe it is fairly close to the current design used in Gutenberg.
![Screenshot 2023-02-09 at 10 56 01](https://user-images.githubusercontent.com/5323259/217779554-4f73c460-7686-4dae-8b8e-9d1d3e6f557d.jpg)

Block settings contains Typography and Dimensions panels. It does not currently contain the split between Styles and Settings seen in example the Post Title block

One can click directly into the "My Site Name. All Rights Reserved." and change the text.

It would be great to do something very similar for a Copyright block added to Gutenberg.

-------

Looking at other Copyright plugins.

@mkaz Marcus has made this Copyright and Date block.
https://wordpress.org/plugins/copyright-block/

No changes can be made as I see. There are no sidebar settings. Insert the block and the text "Copyright © 2023" will be seen.

![Screenshot 2023-02-09 at 11 07 45](https://user-images.githubusercontent.com/5323259/217782300-72d27cd0-d446-49b2-85dc-19cfbc31b156.jpg)

@pbearne Paul has made this Copyright block.
https://github.com/pbiron/shc-copyright/releases/tag/0.9.0

One can click directly into the "© 2023 Copyright Statement." and change Copyright Statement text.
It has a Launch Year setting and an Advanced panel.

![Screenshot 2023-02-09 at 11 13 01](https://user-images.githubusercontent.com/5323259/217783587-07ac0e32-cfee-4d04-a130-b29ec3dc406c.jpg)

EDIT:
I found another Copyright plugin on Github. Editable Copyright Block by Richard.
https://github.com/sidewaysdesign/editable-copyright-block

It looks like this:
![Screenshot 2023-02-09 at 11 35 13](https://user-images.githubusercontent.com/5323259/217788660-1d3e2d27-d2ba-4053-aff4-431ea05dac67.jpg)

One can edit the Prefix (Copyright symbol) and the Suffix (Name of Organization).

---

### Looking at the earlier above comments...

@melchoyce mentions outputting the  site title, ©, and the year. Additionally the author credit.
@BinaryMoon mentions a link to the privacy page.
@mapk mentions license type.
To me this could be a version 2 of a Copyright block. 

A version 1 would really work on adding something very similar to Michelle's plugin mentioned and seen just above in my comment. Are we on a good to go with getting a developer to create version 1?

**Comment by aristath** [comment]
I still don't understand why we **need** this block.
Is it something that users want to import in multiple pages? Then why can't they create a reusable block with a paragraph and their content? The copyright characters etc all exist and can be properly added in any paragraph.
Is it something that users want in all their pages? If so, then why not add a simple paragraph with that content, in their templates?
Why is a separate "copyright" block necessary if all it contains is plain text? If something is missing from a paragraph which prevents this scenario, then I'd rather we enhance the paragraph block to add the missing features instead of creating a new block for this edge case

**Comment by pbiron** [comment]
We don't **need** it, but it's certainly a "nice to have".  At least 2 of the sample blocks that were mentioned in @paaljoachim 's (now closed) ticket are dynamic blocks (including mine): where the year(s) displayed are dynamic, so that a site admin doesn't have to edit the footer template on Jan 1 every year.

In classic themes, I've always included a shortcode to accomplish that dynamic behavior...so it'd be nice to have a core block for block-based themes.  If others decide it's not needed in core, that's fine...I'll just continue to use the block I created ;-)

**Comment by aristath** [comment]
I agree that we need some strings that are dynamic... But I'd prefer to have a mechanism to inject dynamic strings in content - in lieu of custom blocks. I have some ideas on that topic, as soon as we're finished with the WP 6.2 release I plan to make a POC and a suggestion for Gutenberg to include them 👍

**Comment by peroyomas** [comment]
As I said before, I think that an option of insert an inline text token that always displays the CURRENT YEAR is the minimum to have.

**Comment by unscripted** [comment]
Has there been any movement on this? Since it's the new year this is top of mind as we just manually updated several site's copyright year. 

I know there are several plugins that introduce a year block, but I'd love either a core block or dynamic string method.

**Comment by creativecoder** [comment]
The recently published "Bits" proposal is a potential alternative to dynamically display the current year: https://make.wordpress.org/core/2024/06/08/proposal-bits-as-dynamic-tokens/

**Comment by t-hamano** [comment]
Since the formatting you want to use varies from person to person, it's difficult to cover all options in a single block.

I suggest using Block Bindings API instead. Below is an example. 

Register source:

```php
function register_copyright_year_source() {
	register_block_bindings_source(
		'my-plugin/copyright',
		array(
			'label'              => __( 'Copyright Year', 'gutenberg' ),
			'get_value_callback' => function () {
				return sprintf(
					'&copy; %1$s %2$s. All Right Reserved.',
					wp_date( 'Y' ),
					get_bloginfo( 'name' )
				);
			},
		)
	);
}
add_action( 'init', 'register_copyright_year_source' );
```

Bind the source to a paragraph block:

```html
<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"my-plugin/copyright"}}}} -->
<p></p>
<!-- /wp:paragraph -->
```
